fuzz/README.md

```diff
@@ -0,0 +1,110 @@
+# Fuzzing
+
+`simplicity` has a fuzzing harness setup for use with honggfuzz.
+
+To run the fuzz-tests as in CI -- briefly fuzzing every target -- simply
+run
+
+    ./fuzz.sh
+
+in this directory.
+
+You need to install the Rust crate honggfuzz-rs.
+
+To build the crate you must have libunwind on your system, as well as
+libopcodes and libbfd from binutils **2.38** on your system. The most
+recently-released binutils 2.39 has changed their API in a breaking way.
+
+Then you can install it via cargo
+
+    cargo install honggfuzz
+
+On Nix, the installation is automated by using the provided nix shell
+
+    nix-shell
+
+Then run fuzz.sh as above.
+
+# Long-term fuzzing
+
+To see the full list of targets, the most straightforward way is to run
+
+    listTargetNames
+
+You may need to source the fuzz utils, unless you are already in the nix shell
+
+    source ./fuzz-util.sh
+
+To run each target for an hour, run
+
+    ./cycle.sh
+
+This script uses the `chrt` utility to try to reduce the priority of the
+jobs. If you would like to run for longer, the most straightforward way
+is to edit `cycle.sh` before starting. To run the fuzz-tests in parallel,
+you will need to implement a custom harness.
+
+To run a single target indefinitely, run
+
+    cargo hfuzz run <target>
+
+Kill the process when you are done.
+
+# Adding fuzz tests
+
+All fuzz tests can be found in the `fuzz_target/` directory. Adding a new
+one is as simple as copying an existing one and editing the `do_test`
+function to do what you want.
+
+If your test clearly belongs to a specific crate, please put it in that
+crate's directory. Otherwise you can put it directly in `fuzz_target/`.
+
+If you need to add dependencies, edit the file `generate-files.sh` to add
+it to the generated `Cargo.toml`.
+
+Once you've added a fuzztest, regenerate the `Cargo.toml` and CI job by
+running
+
+    ./generate-files.sh
+
+Then to test your fuzztest, run
+
+    ./fuzz.sh <target>
+
+If it is working, you will see a rapid stream of data for many seconds
+(you can hit Ctrl+C to stop it early). If not, you should quickly see
+an error.
+
+# Reproducing Failures
+
+If fuzz.sh or cycle.sh fail, they will exit with a summary which looks something like
+
+```
+...
+ fuzzTarget      : hfuzz_target/x86_64-unknown-linux-gnu/release/decode_natural
+CRASH:
+DESCRIPTION:
+ORIG_FNAME: 00000000000000000000000000000000.00000000.honggfuzz.cov
+FUZZ_FNAME: hfuzz_workspace/decode_natural/SIGABRT.PC.7ffff7e42adc.STACK.1a1d230819.CODE.-6.ADDR.0.INSTR.mov____%eax,%ebx.fuzz
+...
+=====================================================================
+fff400610004
+```
+
+If you run `cargo hfuzz` manually, run the following to see the report
+
+    checkReport <target>
```

Comment:
In 08801420c213b16598eb0d5926c7173ce6f0fd4d:

If you are telling people to run this then they need to source the utils script first.